A couple fixes (#15)
canonical/cassandra-operator: docs/how-to/encryption.md

> # Managing Encryption

Reviewer: Suggestion: Please use sentence case for headings, as per our [style guide](https://docs.ubuntu.com/styleguide/en#headings)

```suggestion
# Managing encryption
```

The Apache Cassandra charm has encryption enabled by default.
All units within a cluster share the same CA certificate file, but each unit has a distinct private key.

> This charm implements the **Requirer** side of the [`tls-certificates/v1`](https://github.com/canonical/charm-relation-interfaces/blob/main/interfaces/tls_certificates/v1/README.md) relation. Therefore, any charm implementing the **Provider** side can be used.

Reviewer: Suggestion: Avoid using the inline code blocks syntax in links text and consider replacing the link with this one - https://charmhub.io/integrations/tls-certificates

```suggestion
This charm implements the **Requirer** side of the [tls-certificates](https://charmhub.io/integrations/tls-certificates) relation. Therefore, any charm implementing the **Provider** side can be used.
```

To enable TLS encryption, you must first deploy a TLS certificates Provider charm.

> ---

Reviewer: Suggestion: Drop this redundant line.

```suggestion
```

> ## Deploy a TLS Provider Charm

Reviewer: Suggestion: Here and further, use sentence case for headings.

For testing purposes, you can use the `self-signed-certificates` charm.
However, this setup is **not recommended** for production clusters.

Deploy the `self-signed-certificates` charm with:

```shell
# Deploy the TLS charm
juju deploy self-signed-certificates --channel=edge

# Configure the CA common name
juju config self-signed-certificates ca-common-name="Test CA"

> ```

Reviewer: Suggestion: Remove the comments from the code block and split the commands with normal text, introducing the second command.

> For an overview of available TLS certificate Provider charms and guidance on choosing the right one for your use case, see [this guide](https://charmhub.io/topics/security-with-x-509-certificates).

Reviewer: Suggestion: Let's avoid "this" in a link text and make it more concrete.

```suggestion
For an overview of available TLS certificate Provider charms and guidance on choosing the right one for your use case, see the [Security with X.509 certificates](https://charmhub.io/topics/security-with-x-509-certificates) page.
```

## Relate the Charms

To enable **peer-to-peer TLS encryption**, relate Cassandra to the `:peer-certificates` endpoint:

```shell

> juju relate <tls-certificates>:certificates cassandra:peer-certificates

Reviewer: Chore: We should replace `relate` with the `integrate` command.

```

To enable **client-to-node TLS encryption**, relate Cassandra to the `:client-certificates` endpoint:

```shell
juju relate <tls-certificates>:certificates cassandra:client-certificates
```

where `<tls-certificates>` is the name of the deployed TLS certificates Provider charm.

## Connect to the Cluster

Authentication is enabled by default.
To retrieve the password for the default `operator` user:

```shell
juju show-secret --reveal "cassandra-peers.<application name>.app" --format json \
  | jq -r '.[].content.Data."operator-password"'
```

> ### Verifying Client TLS

Reviewer: Suggestion: Since it's a how-to guide, lets be more imperatiove and drop -ing here and later.

```suggestion
### Verify client TLS
```

# Managing encryption

The Apache Cassandra charm has encryption enabled by default.
All units within a cluster share the same CA certificate file, but each unit has a distinct private key.

This charm implements the **Requirer** side of the [tls-certificates](https://charmhub.io/integrations/tls-certificates) relation. Therefore, any charm implementing the **Provider** side can be used.
To enable TLS encryption, you must first deploy a TLS certificates Provider charm.

> ## Deploy a tls provider charm

Reviewer: ```suggestion
## Deploy a TLS provider charm
```

Deploy the `self-signed-certificates` charm with:

```shell
juju deploy self-signed-certificates --channel=edge
```

Configure the CA common name:

```shell
juju config self-signed-certificates ca-common-name="Test CA"
```

For an overview of available TLS certificate Provider charms and guidance on choosing the right one for your use case, see [this guide](https://charmhub.io/topics/security-with-x-509-certificates).

## Relate the charms

To enable **peer-to-peer TLS encryption**, relate Cassandra to the `:peer-certificates` endpoint:

```shell
juju integrate <tls-certificates>:certificates cassandra:peer-certificates
```

To enable **client-to-node TLS encryption**, relate Cassandra to the `:client-certificates` endpoint:

```shell
juju integrate <tls-certificates>:certificates cassandra:client-certificates
```

where `<tls-certificates>` is the name of the deployed TLS certificates Provider charm.

## Connect to the cluster

Authentication is enabled by default.
To retrieve the password for the default `operator` user:

```shell
juju show-secret --reveal "cassandra-peers.<application name>.app" --format json \
  | jq -r '.[].content.Data."operator-password"'
```

> ### Verify client tls

Reviewer: ```suggestion
### Verify client TLS
```

Connection error: ('Unable to connect to any servers',
  {'10.166.144.168:9042': ConnectionResetError(104, 'Connection reset by peer')})
```

And in the Cassandra logs you will see:

```
WARN  [epollEventLoopGroup-5-6] ... SSLException in client networking with peer /10.166.144.168:42604
io.netty.handler.ssl.NotSslRecordException: not an SSL/TLS record
```

This confirms that Cassandra requires a secure TLS connection.

### Retrieving client certificates

Fetch the client private key and signed certificate from a unit:

```shell
juju ssh <unit-name> "cat /var/snap/charmed-cassandra/current/etc/cassandra/tls/client-private.key" > ./client.key
juju ssh <unit-name> "cat /var/snap/charmed-cassandra/current/etc/cassandra/tls/client-unit.pem" > ./client.pem
```

### Configuring `cqlsh`

To properly use these files, create a `cqlshrc` configuration file for `cqlsh`:

```ini
[authentication]
username = operator
password = <password>

[connection]
hostname = <unit-ip>
port = 9042

[ssl]
certfile = ./client.pem
validate = true
userkey = ./client.key
usercert = ./client.pem
```

Connect to Cassandra with:

```shell
cqlsh --ssl --cqlshrc=./cqlshrc
```

> ## Disabling tls

Reviewer: ```suggestion
## Disabling TLS
```
